dbms/src/Storages/System/CMakeLists.txt

```diff
@@ -12,7 +12,7 @@ include(${ClickHouse_SOURCE_DIR}/cmake/dbms_glob_sources.cmake)
 add_headers_and_sources(storages_system .)
 list (APPEND storages_system_sources ${CONFIG_BUILD})
 add_library(clickhouse_storages_system ${storages_system_headers} ${storages_system_sources})
-target_link_libraries(clickhouse_storages_system PRIVATE dbms common string_utils clickhouse_common_zookeeper clickhouse_parsers)
+target_link_libraries(clickhouse_storages_system PRIVATE dbms common string_utils clickhouse_common_zookeeper clickhouse_parsers ${JEMALLOC_LIBRARIES})
```

Comment:
I recommend to automatically disable Jemalloc if the build is shared.
At least it is safe... and there is no use for shared build except for faster build on dev machines.

Reply:
It looks like there was attempt to solve the issue here https://github.com/ClickHouse/ClickHouse/blob/master/dbms/src/Common/malloc.cpp#L4

But it compiles into object lib and never used https://github.com/ClickHouse/ClickHouse/blob/master/dbms/CMakeLists.txt#L255